multi-agent/lib/mas/resources/service.py

```
@@ -15,40 +19,70 @@
     and delegates storage to ResourcesRegistry.
+
+    Built-in-awareness (is this resource a built-in? what's its
+    visibility? what's the caller's config overlay?) is never decided
+    here directly — it's delegated through the small helper methods on
+    the injected ``BuiltinResourceService`` (``self._builtin``), which is
+    the sole owner of the ``ResourceOwnership``/``ResourceVisibility``
+    concepts. This class carries zero knowledge of those enums; see
+    ``resources.md`` for the full rationale.
     """
 
     def __init__(
```

Ruff (0.16.1)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)